### PR TITLE
[language-platform] move policies’s background jobs to its own layer

### PR DESCRIPTION
**Pull request by Numbers88s**
Move policies’s background jobs to its own layer

## Test plan
Updated and ran all the tests.

### CODE REVIEW

File: internal/codeintel/policies/init.go [hunk 1]
```diff
@@ -31,13 +32,11 @@ type serviceDependencies struct {
...
 	store := store.New(deps.db, scopedContext("store"))
+	backgroundJob := background.New(scopedContext("background"))
 
-	return newService(
-		store,
-		deps.uploadSvc,
-		deps.gitserver,
-		scopedContext("service"),
-	), nil
+	svc := newService(store, deps.uploadSvc, deps.gitserver, backgroundJob, scopedContext("service"))
```
[the remark this answers is not in the document]
Comment (Numbers88s) [reply internal/codeintel/policies/init.go:37]: cause it fits on one line. Doesn't seem that long to me.

File: internal/database/migration/stitch/stitch_test.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ func TestMain(m *testing.M) {
 // v3.38.0 -> privileged migrations introduced
 
 func TestStitchFrontendDefinitions(t *testing.T) {
+	t.Skip()
```
Comment (efritz) [inline internal/database/migration/stitch/stitch_test.go:37]: 👮‍♂️ HEY

**Review by efritz — CHANGES_REQUESTED** [review]

File: internal/codeintel/policies/init.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (efritz) [reply internal/codeintel/policies/init.go:37]: Makes differs harder to read when we collapse it, or when we have to expand it again in the future.

File: internal/database/migration/stitch/stitch_test.go (hunk 2 above)
Comment (Numbers88s) [reply internal/database/migration/stitch/stitch_test.go:37]: oh snap. Sorry!

File: internal/codeintel/policies/init.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Numbers88s) [reply internal/codeintel/policies/init.go:37]: oh cool cool.  I guess it doesn't bother me, but I can split it if you want. I find it harder to read when I'm coding _shrug_

[the remark this answers is not in the document]
Comment (efritz) [reply internal/codeintel/policies/init.go:37]: You can leave it as-is

File: internal/database/migration/stitch/stitch_test.go (hunk 2 above)
Comment (efritz) [reply internal/database/migration/stitch/stitch_test.go:37]:
```suggestion
```

**Review by efritz — APPROVED** [review]

**Pull request merged**